community/cypher/compatibility-suite/src/main/java/cypher/feature/parser/CypherParametersCreator.java

```diff
@@ -0,0 +1,224 @@
+    {
+        return workload.pop();
+    }
+
+    @Override
+    public void enterInteger( FeatureResultsParser.IntegerContext ctx )
+    {
+        workload.push( Long.valueOf( ctx.getText() ) );
+    }
+
+    @Override
+    public void enterNullValue( FeatureResultsParser.NullValueContext ctx )
+    {
+        workload.push( null );
+    }
+
+    @Override
+    public void enterFloatingPoint( FeatureResultsParser.FloatingPointContext ctx )
+    {
+        String text = ctx.getText();
+        if ( text.contains( INFINITY ) )
+        {
+            workload.push( Double.parseDouble( text + "inity" ) );
+        }
+        else
+        {
+            workload.push( Double.parseDouble( text ) );
+        }
+    }
+
+    @Override
+    public void enterBool( FeatureResultsParser.BoolContext ctx )
+    {
+        workload.push( Boolean.valueOf( ctx.getText() ) );
+    }
+
+    @Override
+    public void enterString( FeatureResultsParser.StringContext ctx )
+    {
+        String text = ctx.getText();
+        String substring = text.substring( 1,
+                text.length() - 1 ); // remove wrapping quotes -- because I can't get the parser rules correct :(
+        String escaped = substring.replace( "\\'", "'" ); // remove escaping backslash -- see above comment
+        workload.push( escaped );
+    }
+
+    @Override
+    public void enterList( FeatureResultsParser.ListContext ctx )
+    {
+        listCounters.push( 0 );
+    }
+
+    @Override
+    public void enterListElement( FeatureResultsParser.ListElementContext ctx )
+    {
+        listCounters.push( listCounters.pop() + 1 );
+    }
+
+    @Override
+    public void exitList( FeatureResultsParser.ListContext ctx )
+    {
+        // Using a Deque here in order to be able to prepend
+        Deque<Object> temp = new LinkedList<>();
+        int counter = listCounters.pop();
+        for ( int i = 0; i < counter; ++i )
+        {
+            temp.addFirst( workload.pop() );
+        }
+        workload.push( temp );
+    }
+
+    @Override
+    public void enterPropertyMap( FeatureResultsParser.PropertyMapContext ctx )
+    {
+        mapCounters.push( 0 );
+    }
+
+    @Override
+    public void enterKeyValuePair( FeatureResultsParser.KeyValuePairContext ctx )
+    {
+        mapCounters.push( mapCounters.pop() + 1 );
+    }
+
+    @Override
+    public void exitPropertyMap( FeatureResultsParser.PropertyMapContext ctx )
+    {
+        int counter = mapCounters.pop();
+        Map<String,Object> map = new HashMap<>();
+        for ( int i = 0; i < counter; ++i )
+        {
+            Object value = workload.pop();
+            String key = workload.pop().toString();
+            map.put( key, value );
+        }
+        workload.push( map );
+    }
+
+    @Override
+    public void enterPropertyKey( FeatureResultsParser.PropertyKeyContext ctx )
+    {
+        workload.push( ctx.getText() );
+    }
+
+
+    // TODO: Solve question about graph elements as parameter values
```

Comment:
What is to be done about this? And should all the jazz below here really be there?